clippy_config/src/conf.rs

```diff
@@ -526,6 +526,25 @@ define_Conf! {
     /// A list of paths to types that should be treated as if they do not contain interior mutability
     #[lints(borrow_interior_mutable_const, declare_interior_mutable_const, ifs_same_cond, mutable_key_type)]
     ignore_interior_mutability: Vec<String> = Vec::from(["bytes::Bytes".into()]),
+    /// Whether to suggest reordering constructor fields when initializers are present.
+    ///
+    /// Note that such suggestions are not applied automatically with `--fix`. The following example
+    /// [due to @ronnodas] shows why. Swapping the fields in the constructor produces incompilable code:
+    ///
+    /// ```rust
+    /// struct MyStruct {
+    ///     vector: Vec<u32>,
+    ///     length: usize
+    /// }
+    /// fn main() {
+    ///     let vector = vec![1,2,3];
+    ///     MyStruct { length: vector.len(), vector};
+    /// }
+    /// ```
+    ///
+    /// [due to @ronnodas]: https://github.com/rust-lang/rust-clippy/issues/11846#issuecomment-1820747924
+    #[lints(inconsistent_struct_constructor)]
+    initializer_suggestions: bool = false,
```

Comment:
> What do you think about naming this `warn_inconsistent_struct_field_initializers`? IMO "suggestions" isn't so accurate anymore after the last change since whether this is enabled or not has no effect on the suggestion now

Could I suggest `lint_inconsistent_struct_field_initializers`, or some other verb besides `warn`?

camsteffen once [pointed out to me](https://github.com/rust-lang/rust-clippy/pull/7743#discussion_r721396067) that whether the lint actually warns is configurable.

Reply:
Yeah that works too. `warn` seemed consistent with other similar named configs (there are two that start with `warn_*`), but I agree with you that it's a bit of a misnomer when someone denies the lint